src/WatchDogs.Persistence.EntityFramework/DataInserter.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using WatchDogs.Contracts;
+
+namespace WatchDogs.Persistence.EntityFramework;
+public class DataInserter
+{
+    private readonly ApplicationDbContext _context;
+
+    public DataInserter(ApplicationDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task InsertTradeDatatoDbAsync(IEnumerable<Trade> data)
+    {
+        foreach (var trade in data)
+        {
+            await _context.Trades.AddAsync(trade);
```

Comment:
Not sure about EF core, but from the C# point of view, it is slow, although correct.

If we had 1000 deals incoming, we would insert deal 1, await its insertion, then insert seal 2, await, etc. You probably sense we could do better.

AddSync returns a Task that can naturally be awaited. Furthermore, class Task has a static method called WhenAll that accepts a collection of tasks and completed when all tasks in the collection are completed. I would try this approach if I were you (you'll learn a great deal)